interpreter/cling/lib/Interpreter/BackendPasses.cpp

```diff
@@ -40,6 +40,48 @@ using namespace cling;
 using namespace clang;
 using namespace llvm;
 
+namespace {
+  class UniqueInitFunctionNamePass
+      : public PassInfoMixin<UniqueInitFunctionNamePass> {
+    // append a suffix to a symbol to make it unique
+    // the suffix is "_cling_module_<module number>"
+    llvm::SmallString<128> add_module_suffix(const StringRef OriginalName,
+                                             const StringRef ModuleName) {
+      llvm::SmallString<128> NewFunctionName;
+      NewFunctionName.append(ModuleName);
+      NewFunctionName.append("_");
+
+      for (size_t i = 0; i < NewFunctionName.size(); ++i) {
+        // Replace everything that is not [a-zA-Z0-9._] with a _. This set
+        // happens to be the set of C preprocessing numbers.
+        if (!isPreprocessingNumberBody(NewFunctionName[i]))
+          NewFunctionName[i] = '_';
+      }
+
+      return NewFunctionName;
+    }
+
+    // make static initialization function names (__cxx_global_var_init) unique
+    bool runOnFunction(Function& F, const StringRef ModuleName) {
+      if (F.hasName() && F.getName().startswith("__cxx_global_var_init")) {
```

Comment:
`llvm::StringRef::startswith` is deprecated, we should call `starts_with`.

(There are a couple of other uses of deprecated functions in our downstream patches and `core/metacling`. They are on my list of things to fix in January and also remove `-Wno-deprecated-declarations` to make the compiler warn about them.)